lib/libxdp/xsk.c

```diff
@@ -1093,6 +1217,27 @@ int xsk_umem__delete(struct xsk_umem *umem)
 	return 0;
 }
 
+static void xsk_release_xdp_prog(struct xsk_socket *xsk)
+{
+	struct xsk_ctx *ctx = xsk->ctx;
+	int value;
+
+	if (!xsk->ctx->refcnt_map_fd)
+		goto ret;
+
+	value = xsk_decr_prog_refcnt(ctx->refcnt_map_fd);
+	if (value < 0)
+		pr_warn("Error occurred when decrementing xsk XDP prog refcount: %d, please detach program yourself\n",
+			value);
```

Comment:
`strerror()` here, maybe? Also, maybe just "not detaching program" instead of "please detach yourself" ? :)